pkg/sharing/files.go

```diff
@@ -3,36 +3,77 @@ package sharing
 import (
 	"github.com/cozy/cozy-stack/pkg/consts"
 	"github.com/cozy/cozy-stack/pkg/couchdb"
+	"github.com/cozy/cozy-stack/pkg/crypto"
 	"github.com/cozy/cozy-stack/pkg/instance"
 	"github.com/cozy/cozy-stack/pkg/vfs"
 )
 
+// MakeXorKey generates a key for transforming the file identifiers
+func MakeXorKey() []byte {
+	random := crypto.GenerateRandomBytes(8)
+	result := make([]byte, 2*len(random))
+	for i, val := range random {
+		result[2*i] = val & 0xf
+		result[2*i+1] = val >> 4
+	}
+	return result
+}
+
+// XorID transforms the identifier of a file to a new identifier, in a
+// reversible way: it makes a XOR on the hexadecimal characters
+func XorID(id string, key []byte) string {
```

Comment:
If the `dir_id` of a file is `io.cozy.files.root-dir`, actually few characters will be XORed. I'm not sure if this is problematic, though.

Reply:
What I aim is to not call XorID for root-dir and trash-dir. I'm not sure about the details of how to do that yet.